sdk/identity/src/token_credentials/client_certificate_credentials.rs

```diff
@@ -138,12 +140,12 @@ struct AadTokenResponse {
 fn get_encoded_cert(cert: &X509) -> azure_core::Result<String> {
     Ok(format!(
         "\"{}\"",
-        base64::encode(cert.to_pem().map_err(|_| openssl_error())?)
+        base64::encode(cert.to_pem().map_err(openssl_error)?)
     ))
 }
 
-fn openssl_error() -> azure_core::error::Error {
-    Error::message(ErrorKind::Credential, "Openssl decode error")
+fn openssl_error(err: ErrorStack) -> azure_core::error::Error {
+    Error::new(ErrorKind::DataConversion, err)
```

Comment:
Are we sure `DataConversion` is a better choice than `Credential`?

Reply:
Thus far, we've used DataConversion is primarily used to indicate parsing errors into and out of the REST endpoints.

Reply:
I wasn't sure what to use. This is an error happening in data conversion from pem so i thought it would be more appropriate. i can update it to credential. Don't have a strong stance for it.

Reply:
I don't think it matters too much, but for now, I would keep it as `ErrorKind::Credential`. We may want to revisit these `ErrorKind`s in general, as it's becoming clear when they're useful and when they're just boiler plate...

Reply:
fixed.